### PR TITLE
Update chalk to 0.28.0

### PR DESCRIPTION
**Pull request by vandenheuvel**

### CODE REVIEW

File: crates/hir_ty/src/traits/chalk.rs [hunk 1]
```diff
@@ -413,30 +417,36 @@ pub(crate) fn trait_datum_query(
...
+        "fn" => Fn,
+        "unsize" => Unsize,
+        "coerce_unsized" => CoerceUnsized,
         _ => return None,
     })
 }
 
-fn lang_attr_from_well_known_trait(attr: WellKnownTrait) -> &'static str {
+const fn lang_attr_from_well_known_trait(attr: WellKnownTrait) -> &'static str {
+    use chalk_solve::rust_ir::WellKnownTrait::*;
```
Comment (lnicola) [inline crates/hir_ty/src/traits/chalk.rs:437]: Can you leave these as before? We have [this page](https://rust-analyzer.github.io/metrics/) and this import will pollute the results (see https://github.com/rust-analyzer/rust-analyzer/issues/1165).

Comment (matklad) [reply crates/hir_ty/src/traits/chalk.rs]: Yup, we also have "avoid local enum imports" in the style guide: https://github.com/rust-analyzer/rust-analyzer/blob/master/docs/dev/style.md#import-style

**Comment by matklad** [comment]
bors r+

Thanks!

**Comment by matklad** [comment]
Hm, seems likely that this introduced a crash on analysis-stats for rust-analyzer itself: https://github.com/rust-analyzer/rust-analyzer/runs/1174103663?check_suite_focus=true#step:9:18

**Comment by matklad** [comment]
cc @flodiebold

**Comment by flodiebold** [comment]
I'll try to look into it today or tomorrow.

**Comment by matklad** [comment]
Thanks!

**Comment by flodiebold** [comment]
Apparently it's been fixed already: https://github.com/rust-lang/chalk/pull/613

**Comment by matklad** [comment]
Splendid! I wonder if we should add a test for this?

**Comment by flodiebold** [comment]
Hmm, a test for this specific case will be added to Chalk, so I don't know whether a specific test case will help here, and I don't know if we can write a more general test...

**Pull request merged**